### DISCUSSION TITLE
Clarification on Spaces

### DISCUSSION THREAD

**Discussion by zbhoy** [rxhanson/RectanglePro-Community#435] [Q&A] [answered]
Hi there, 

I think I have the answer to this but I just want clarification. 

I work in 3 modes.

Mode 1 - Laptop only
- Multiple spaces with app on each 

Mode 2 - Laptop hooked up to ultrawide monitor
- Both displays used
- Multiple Spaces on each one with apps in each one 

Mode 3 - Laptop hooked up to UW monitor, laptop in clamshell
- Only UW used 
- Multiple spaces with apps in each one 

Does saved layouts enable that when i plug into the UW it arranges the apps into the spaces I want and when i disconnect it then arranges them back into the spaces they were in on laptop only?

I believe it is No because no Spaces API. 

If it is No, is there a way that I can plug in UW and press a shortcut and then it does it?

**Comment by rxhanson** [comment, marked as answer]
Unfortunately, you are correct. Without a Spaces API, I have no way to perform any action on any window on a Space that is not visible. What we're left with on Rectangle Pro is creating App Layouts and manually triggering them or triggering them when displays are connected/disconnected. It's on my backlog to allow you to specify layouts for when specific displays are connected/disconnected.

**Reply by zbhoy** [reply]
Understood. So even with the display connected and disconnected options it wouldn't be able to put the apps into different spaces? Would just lay them out in the first space or whatever space they are put in?

**Reply by rxhanson** [reply]
Correct :(

**Comment by ascruggs** [comment]
There appears to be an opensource project that does this:  https://ianyh.com/amethyst/ .  @rxhanson would you be open to implementing a solution that doesn't use an API?  (I'm guessing it uses a hotkeys hack).

**Reply by ascruggs** [reply]
Just say [your comment last year about this](https://github.com/rxhanson/RectanglePro-Community/discussions/8#discussioncomment-5412150).  Just curious if your position has changed?  I really appreciate the Rectangle is a fully baked product and understand your hesitancy.  This just seems to be such a pain point.  Perhaps add it with an 'experimental' flag?

**Reply by rxhanson** [reply]
In Rectangle Pro you can configure a shortcut to move a window next/prev space. It's still a bit of a hack, and not something I want to add into Rectangle.

**Reply by ascruggs** [reply]
Makes sense.  As I pro user, I wish it worked with app layouts.  When I plug in my external monitor, I want my apps to move to the right space and be arranged how I configured it.  Is there a way to accomplish this as a pro user?

**Reply by rxhanson** [reply]
Unfortunately not. Sorry!
